.github/workflows/codeql-build.yml

```diff
@@ -1,37 +1,23 @@
 name: "CodeQL Analysis"
 
+# Only trigger, when the build workflow succeeded
 on:
-  push:
-  pull_request:
-    branches:
-      - main
-
+  workflow_run:
+    workflows: 
+      - Build, Test, and Run
+    types:
+      - completed
+    branches: 
+      - '**'
 env:
   SIMULATION: native
   ENABLE_UNIT_TESTS: true
   OMIT_DEPRECATED: true
   BUILDTYPE: release
 
-jobs:
-  #Checks for duplicate actions. Skips push actions if there is a matching or duplicate pull-request action. 
-  check-for-duplicates:
-    runs-on: ubuntu-latest
-    # Map a step output to a job output
-    outputs:
-      should_skip: ${{ steps.skip_check.outputs.should_skip }}
-    steps:
-      - id: skip_check
-        uses: fkirc/skip-duplicate-actions@master
-        with:
-          concurrent_skipping: 'same_content'
-          skip_after_successful_duplicate: 'true'
-          do_not_skip: '["pull_request", "workflow_dispatch", "schedule"]'
-
-      
+jobs:      
```

Comment:
Are we sure this won't run twice without the check?

Reply:
CodeQL is only triggered on workflow_run, but I can keep the check there for now.